cli/azd/pkg/infra/provisioning/manager.go

```diff
@@ -86,12 +97,146 @@ func (m *Manager) Deploy(ctx context.Context) (*DeployResult, error) {
+			op.StorageAccount,
+			op.FileShareName,
+			op.Path); err != nil {
+			return nil, fmt.Errorf("error binding mount: %w", err)
+		}
+		m.console.MessageUxItem(ctx, &ux.DisplayedResource{
+			Type:  fileShareUploadOperation,
+			Name:  op.Description,
+			State: ux.SucceededState,
+		})
+	}
+
 	// make sure any spinner is stopped
 	m.console.StopSpinner(ctx, "", input.StepDone)
 
 	return deployResult, nil
 }
 
+const (
+	fileShareUploadOperation string = "FileShareUpload"
+	azdOperationsFileName    string = "azd.operations.yaml"
+)
+
+type azdOperation struct {
+	Type        string
+	Description string
+	Config      any
+}
+
+type azdOperationFileShareUpload struct {
+	Description    string
+	StorageAccount string
+	FileShareName  string
+	Path           string
+}
+
+type azdOperationsModel struct {
+	Operations []azdOperation
+}
+
+func azdOperations(infraPath string, env environment.Environment) (azdOperationsModel, error) {
+	// Check if the file exists
+	path := filepath.Join(infraPath, azdOperationsFileName)
+	if _, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			// File does not exist
+			return azdOperationsModel{}, nil
+		}
+		// Error occurred while checking file status
+		return azdOperationsModel{}, err
+	}
+
+	// Read the file
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return azdOperationsModel{}, err
+	}
+
+	// resolve environment variables
+	expString := osutil.NewExpandableString(string(data))
+	evaluated, err := expString.Envsubst(env.Getenv)
+	if err != nil {
+		return azdOperationsModel{}, err
+	}
+	data = []byte(evaluated)
+
+	// Unmarshal the file into azdOperationsModel
+	var operations azdOperationsModel
+	err = yaml.Unmarshal(data, &operations)
+	if err != nil {
+		return azdOperationsModel{}, err
+	}
+
+	return operations, nil
+}
+
+func azdFileShareUploadOperations(infraPath string, env environment.Environment) ([]azdOperationFileShareUpload, error) {
+	model, err := azdOperations(infraPath, env)
+	if err != nil {
+		return nil, err
+	}
+
+	var fileShareUploadOperations []azdOperationFileShareUpload
+	for _, operation := range model.Operations {
+		if operation.Type == fileShareUploadOperation {
+			var fileShareUpload azdOperationFileShareUpload
+			bytes, err := json.Marshal(operation.Config)
+			if err != nil {
+				return nil, err
+			}
+			err = json.Unmarshal(bytes, &fileShareUpload)
+			if err != nil {
+				return nil, err
+			}
+			fileShareUpload.Description = operation.Description
+			fileShareUploadOperations = append(fileShareUploadOperations, fileShareUpload)
+		}
+	}
+	return fileShareUploadOperations, nil
+}
+
+func bindMountOperation(
+	ctx context.Context,
+	fileShareService storage.FileShareService,
+	cloud, subId, storageAccount, fileShareName, source string) error {
+
+	shareUrl := fmt.Sprintf("https://%s.file.%s/%s", storageAccount, cloud, fileShareName)
+
+	return filepath.WalkDir(source, func(path string, info fs.DirEntry, err error) error {
```

Comment:
Given that `fileShareService.UploadPath` uploads the entire tree rooted at path, do we need the `filepath.WalkDir` here?

Reply:
`fileShareService.UploadPath` is actually uploading just one file :).

I have update it to upload all files now and changed this code.